crates/aptos/src/move_tool/bytecode.rs

```diff
@@ -0,0 +1,306 @@
+        let no_loc = Spanned::unsafe_no_loc(()).loc;
+        let module: CompiledModule;
+        let script: CompiledScript;
+        let bytecode = if self.is_script {
+            script = CompiledScript::deserialize(&bytecode_bytes)
+                .context("Script blob can't be deserialized")?;
+            BinaryIndexedView::Script(&script)
+        } else {
+            module = CompiledModule::deserialize(&bytecode_bytes)
+                .context("Module blob can't be deserialized")?;
+            BinaryIndexedView::Module(&module)
+        };
+
+        let mut source_mapping = if let Some(s) = source_map {
+            SourceMapping::new(s, bytecode)
+        } else {
+            SourceMapping::new_from_view(bytecode, no_loc)
+                .context("Unable to build dummy source mapping")?
+        };
+
+        if let Some(source_code) = source {
+            source_mapping.with_source_code((bytecode_path.display().to_string(), source_code));
+        }
+
+        let mut disassembler = Disassembler::new(source_mapping, disassembler_options);
+
+        if let Some(file_path) = &self.code_coverage_path {
+            disassembler.add_coverage_map(
+                CoverageMap::from_binary_file(file_path)
+                    .map_err(|_err| {
+                        CliError::UnexpectedError("Unable to read from file_path".to_string())
+                    })?
+                    .to_unified_exec_map(),
+            );
+        }
+
+        disassembler
+            .disassemble()
+            .map_err(|_err| CliError::UnexpectedError("Unable to disassemble".to_string()))
+    }
+
+    fn decompile(&self, bytecode_path: &Path) -> Result<String, CliError> {
+        let exe = self.get_revela_path()?;
+        let to_cli_error = |e| CliError::IO(exe.display().to_string(), e);
+        let mut cmd = Command::new(exe.as_path());
+        cmd.arg(format!("--bytecode={}", bytecode_path.display()));
+        if self.is_script {
+            cmd.arg("--script");
+        }
+        let out = cmd.output().map_err(to_cli_error)?;
+        if out.status.success() {
+            String::from_utf8(out.stdout).map_err(|_| {
+                CliError::UnexpectedError(
+                    "output generated by decompiler is not valid utf8".to_owned(),
+                )
+            })
+        } else {
+            Err(CliError::UnexpectedError(format!(
+                "decompiler exited with status {}: {}",
+                out.status,
+                String::from_utf8(out.stderr).unwrap_or_default()
+            )))
+        }
+    }
+
+    fn get_revela_path(&self) -> CliTypedResult<PathBuf> {
+        if let Ok(path) = std::env::var(REVELA_EXE_ENV) {
+            Ok(PathBuf::from(path))
+        } else if let Some(path) = pathsearch::find_executable_in_path(REVELA_EXE) {
+            Ok(path)
+        } else {
+            Err(CliError::CommandArgumentError(format!(
+                "Cannot locate the decompiler executable. \
```

Comment:
@bamobah @gregnazario: please check the TODO below and advice were to point to.

Reply:
If there isn't an installation page for the revala exe, we can just leave TODO, and we can fill it in when we get the script / a location for instructions on aptos.dev / revala's repo.